lib/AST/NameLookup.cpp

```diff
@@ -2235,6 +2235,23 @@ SuperclassDeclRequest::evaluate(Evaluator &evaluator,
   return nullptr;
 }
 
+ArrayRef<ProtocolDecl *>
+InheritedProtocolsRequest::evaluate(Evaluator &evaluator,
+                                    ProtocolDecl *PD) const {
+  llvm::SmallVector<ProtocolDecl *, 2> result;
+  SmallPtrSet<const ProtocolDecl *, 2> known;
+  known.insert(PD);
+  bool anyObject = false;
```

Comment:
This is doing the same calculation as ProtocolRequiresClassRequest, except it's keeping a different part of the result. Perhaps we can combine the two into a single request that evaluates into an std::pair<>, and then ::requiresClass() and ::getInheritedProtocols() can kick off the combined request and project a different component of the result?